### PR TITLE
Add support for the cc0 license

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
So that we can add cc0 to argo.  Ref https://github.com/sul-dlss/argo/issues/2617

### CODE REVIEW

File: app/services/cocina/to_fedora/access.rb [hunk 1]
```diff
@@ -20,70 +20,19 @@ def initialize(item, access)
...
         # See https://github.com/sul-dlss/dor-services/blob/main/lib/dor/datastreams/rights_metadata_ds.rb
-        Dor::RightsMetadataDS.upd_rights_xml_for_rights_type(item.rightsMetadata.ng_xml, Rights.rights_type(access))
+        Dor::RightsMetadataDS.upd_rights_xml_for_rights_type(rightsMetadata.ng_xml, Rights.rights_type(access))
         # This invalidates the dra_object, which is necessary if re-mapping.
-        item.rightsMetadata.content = item.rightsMetadata.ng_xml.to_s
-        item.rightsMetadata.copyright = access.copyright if access.copyright
-        item.rightsMetadata.use_statement = access.useAndReproductionStatement if access.useAndReproductionStatement
-        lookup_and_assign_license! if access.license
-        item.rightsMetadata.ng_xml_will_change!
+        rightsMetadata.content = item.rightsMetadata.ng_xml.to_s
```
Comment (mjgiarlo) [inline app/services/cocina/to_fedora/access.rb:25]: Should we also drop the explicit `item.` on the right side of the assignment?

File: app/services/cocina/to_fedora/license.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+        if uri.blank?
+          clear_licenses
+        elsif uri.start_with?('https://creativecommons.org/')
+          assign_creative_commons_license(uri, license)
+        elsif uri.start_with?('http://opendatacommons.org/')
+          assign_open_data_commons_license(uri, license)
+        elsif uri == Cocina::FromFedora::Access::NONE_LICENSE_URI
+          datastream.creative_commons = license.code
+          datastream.open_data_commons.uri = ''
```
Comment (mjgiarlo) [inline app/services/cocina/to_fedora/license.rb:49]: `s/open_data_commons/creative_commons/` ?

File: app/services/cocina/to_fedora/default_rights.rb [hunk 3]
```diff
@@ -5,27 +5,9 @@ module ToFedora
...
-        'https://creativecommons.org/publicdomain/mark/1.0/' => 'pdm',
-        'http://opendatacommons.org/licenses/pddl/1.0/' => 'pddl',
-        'http://opendatacommons.org/licenses/by/1.0/' => 'odc-by',
-        'http://opendatacommons.org/licenses/odbl/1.0/' => 'odc-odbl'
-      }.freeze
-
       def self.write(default_object_rights_ds, default_access)
+        License.update(default_object_rights_ds, default_access.license)
         default_object_rights_ds.default_rights = Rights.rights_type(default_access)
-        default_object_rights_ds.use_license = LICENSE_CODES.fetch(default_access.license)
```
Comment (mjgiarlo) [inline app/services/cocina/to_fedora/default_rights.rb:28]: I *think* our license code (now in the `License` class) is now similar enough to `DefaultObjectRights#use_license=` for this to be OK.

File: config/initializers/rights_metadata_monkeypatch.rb [hunk 4]
```diff
@@ -17,4 +17,40 @@ class RightsMetadataDS < ActiveFedora::OmDatastream
 
     define_template(:use, &:use)
   end
+
+  # Overwrite some definitions on the DefaultObjectRightsDS (used in AdminPolicies)
```
Comment (mjgiarlo) [inline config/initializers/rights_metadata_monkeypatch.rb:21]: What's better than one monkey? TWO MONKEYS, BABY 

:monkey: :monkey: 

:monkey: :monkey: 

:monkey: :monkey: 

:monkey: :monkey: 

:monkey: :monkey: 

:monkey: :monkey:

Comment (jcoyne) [reply config/initializers/rights_metadata_monkeypatch.rb:21]: Monkey patching is back!

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**